pravega-client-examples/src/main/java/io/pravega/example/streamprocessing/ExactlyOnceMultithreadedProcessor.java

```diff
@@ -0,0 +1,300 @@
+            Checkpoint checkpoint = Checkpoint.fromBytes(ByteBuffer.wrap(Files.readAllBytes(checkpointPath)));
+            log.info("Starting from checkpointName={}, positions={}", checkpointName, checkpoint.asImpl().getPositions());
+            builder = builder.startFromCheckpoint(checkpoint);
+            startFromCheckpointName = checkpointName;
+        } else {
+            log.warn("Checkpoint file {} not found. Starting processing from the earliest event.", checkpointRootPath);
+
+            // Create streams.
+            try (StreamManager streamManager = StreamManager.create(controllerURI)) {
+                streamManager.createScope(scope);
+                StreamConfiguration streamConfig = StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.byEventRate(
+                                Parameters.getTargetRateEventsPerSec(),
+                                Parameters.getScaleFactor(),
+                                Parameters.getMinNumSegments()))
+                        .build();
+                streamManager.createStream(scope, inputStreamName, streamConfig);
+                streamManager.createStream(scope, outputStreamName, streamConfig);
+            }
+
+            // Create a reader group that starts from the earliest event.
+            builder = builder.stream(Stream.of(scope, inputStreamName));
+            startFromCheckpointName = null;
+        }
+
+        clientFactory = ClientFactory.withScope(scope, controllerURI);
+        writer = clientFactory.createEventWriter(
+                outputStreamName,
+                new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // Create a reader group manager. It must remain open to allow manual checkpoints to work.
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI);
+
+        final ReaderGroupConfig readerGroupConfig = builder.build();
+        readerGroupName = UUID.randomUUID().toString().replace("-", "");
+        readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig);
+        readerGroup = readerGroupManager.getReaderGroup(readerGroupName);
+
+        initiateCheckpointExecutor = Executors.newScheduledThreadPool(1);
+        performCheckpointExecutor = Executors.newScheduledThreadPool(1);
+        workerExecutor = Executors.newFixedThreadPool(numWorkers);
+    }
+
+    /**
+     * Commit all transactions that are part of a checkpoint.
+     *
+     * @param checkpointName
+     */
+    private void commitTransactions(String checkpointName) {
+        log.info("commitTransactions: BEGIN");
+
+        // Read the contents of all pravega-transactions-worker-XX files.
+        // These files contain the Pravega transaction IDs that must be committed now.
+        Path checkpointDirPath = checkpointRootPath.resolve(checkpointName);
+        List<UUID> txnIds = IntStream
+                .range(0, numWorkers)
```

Comment:
Will it not be an issue if the program is restarted with different number of worked thread count? Ideally the range should reflect as many number of transaction prefix file that we have in the checkpoint directory.

Reply:
Yes this is a problem. I will fix this.

Reply:
Fixed.